aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -178,6 +180,31 @@ module aptos_framework::block {
         );
     }
 
+    ///  Emit a `NewBlockEvent` event. This function will be invoked by write set script directly to generate the
+    ///  new block event for WriteSetPayload.
+    public fun emit_writeset_block_event(framework_signer: &signer) acquires BlockResource {
+        system_addresses::assert_aptos_framework(framework_signer);
+
+        let block_metadata_ref = borrow_global_mut<BlockResource>(@aptos_framework);
+        block_metadata_ref.height = event::counter(&block_metadata_ref.new_block_events);
+
+        let genesis_id = @0x0;
```

Comment:
we should use a different id though, we need to guarantee uniqueness of the block id

Reply:
This could be hard... Is there a way to generate a unique id in Move? Plus do we need for this hash to be a preimage of sth?

Reply:
I think this can just come from the input, we can use 0x1 0x2 etc for all writeset txns

Reply:
Sounds dangerous we need to pass it in manually.. how about add a counter field somewhere? `next_genesis_block_id = 0`

Reply:
How would a counter field help?

Reply:
so each time you generate a new epoch, you can use that as the special block id?